src/Elasticsearch.Net/Configuration/ConnectionConfiguration.cs

```diff
@@ -23,12 +23,34 @@ namespace Elasticsearch.Net
 	/// </summary>
 	public class ConnectionConfiguration : ConnectionConfiguration<ConnectionConfiguration>
 	{
-#if DOTNETCORE
-		private static bool IsCurlHandler { get; } = typeof(HttpClientHandler).Assembly.GetType("System.Net.Http.CurlHandler") != null;
+		/// <summary>
+		/// Detects whether we are running on .NET Core without SocketsHttpHandler existing or being enabled
+		/// If this is true we will set a very restrictive <see cref="DefaultConnectionLimit"/>
+		/// As the old curl based handler is known to bleed TCP connections:
+		/// <para>https://github.com/dotnet/runtime/issues/22366</para>
+		/// </summary>
+        private static bool OnDotNetCoreWithNoSocketsHttpHandlerEnabled

```

Comment:
the double negative is quite hard to mentally parse! I think the logic might be better represented as `SocketsHttpHandlerPresentAndDisabled`, or perhaps renaming to `UsingCurlHandler` and detecting `System.Net.Http.CurlHandler` too, as this implementation will also impose the `Environment.ProcessorCount` connection limit on Windows .NET Core when `System.Net.Http.UseSocketsHttpHandler` is disabled and falls back to `System.Net.Http.WinHttpHandler`